uqcsbot/snailrace.py

```diff
@@ -0,0 +1,97 @@
+import discord, asyncio
+from discord import app_commands, ui
+
+from discord.ext import commands
+from uqcsbot.bot import UQCSBot
+
+import uqcsbot.utils.snailrace_utils as snail
+
+# Trying out Discord buttons for Snail Race Interactions
+class SnailRaceView(discord.ui.View):
+    def __init__(self, raceState: snail.SnailRaceState):
+        super().__init__(timeout=snail.SNAILRACE_OPEN_TIME)
+        self.raceState = raceState
+    
+    async def on_timeout(self):
+        """
+        Called when the view times out. This will deactivate the buttons and
+        begine the race.
+        """
+        for child in self.children:
+            child.disabled = True
+        await self.raceState.open_interaction.edit_original_response(content=snail.SNAILRACE_ENTRY_CLOSE, view=self)
+        await self.raceState.race_start()
+
+    @ui.button(label="Enter Race", style=discord.ButtonStyle.primary)
+    async def button_callback(self, interaction: discord.Interaction, button: discord.ui.Button):
+        if self.raceState.add_racer(interaction.user):
+            await interaction.response.send_message(snail.SNAILRACE_JOIN % interaction.user.mention)
+            return
+        
+        await interaction.response.send_message(snail.SNAILRACE_ALREADY_JOINED % interaction.user.mention)
+        
+
+class SnailRace(commands.Cog):
+    snailrace_group = app_commands.Group(name="snailrace", description="Snail Race Commands")
```

Comment:
Given that it is a single command, is a group needed? I don't know a lot about this, so feel free to overrule this.

Reply:
I've changed it to a single commands `/snailrace` I was thinking of potentially implementing multiple commands for `snailrace` like `scoreboard` etc. But I couldn't be bothered storing the data, so its a maybe later thing to do.